app/models/client.rb

```diff
@@ -321,6 +321,52 @@ def target_id=(value)
     Doi.transfer(client_id: symbol.downcase, target_id: target.id)
   end
 
+  def transfer(options = {})
+
+    if options[:target_id].blank?
+      Rails.logger.error "[Transfer] No target provider provided."
+      return nil
+    end
+
+    target_provider = Provider.where(symbol: options[:target_id]).first
+
+    if target_provider.blank?
+      Rails.logger.error "[Transfer] Provider doesn't exist."
+      return nil
+    end
+    
+    if target_provider.member_type == "consortium"
+      Rails.logger.error "[Transfer] Consortiums cannot have repositories."
+      return nil
+    end
+
+    original_provider = Provider.where(symbol: provider_id).first
+    ## Transfer client
+    update_attribute(:allocator, target_provider.id)
+
+    # These prefixes are used by multiple clients
+    prefixes_to_keep = ["10.4124", "10.4225", "10.4226", "10.4227"]
+
+    # delete all associated prefixes
```

Comment:
Can you explain what you are doing here? Are you deleting all provider prefixes? Do you want to delete only prefix(es) used by the client?

Do you want to check whether a prefix used by the client is also used by other clients of the provider and then not allow the transfer?

Reply:
this has been addressed now